yt_dlp/extractor/microsoftembed.py

```diff
@@ -63,3 +71,217 @@ def _real_extract(self, url):
+            'subtitles': 'count:14',
+        },
+        'params': {'listsubtitles': True},
+    }, {
+        'url': 'https://learn.microsoft.com/en-us/shows/bash-for-beginners',
+        'info_dict': {
+            'id': 'bash-for-beginners',
+            'title': 'Bash for Beginners',
+            'description': 'md5:16a91c07222117d1e00912f0dbc02c2c',
+        },
+        'playlist_count': 20,
+    }]
+
+    def _entries(self, url_base, video_id):
+        skip = 0
+        while True:
+            playlist_info = self._download_json(f'{url_base}&$skip={skip}', video_id, f'Downloading entries {skip}')
+            items = traverse_obj(playlist_info, (
+                'results', ..., 'url', {lambda x: self.url_result(f'https://learn.microsoft.com/en-us{x}')}))
+            yield from items
+            skip += len(items)
+            if skip >= playlist_info['count'] or not items:
+                break
+
+    def _real_extract(self, url):
+        video_type, series, slug = self._match_valid_url(url).groups()
+        video_id = slug or series
+        webpage = self._download_webpage(url, video_id)
+
+        metainfo = {
+            'title': self._og_search_title(webpage),
+            'description': self._og_search_description(webpage),
+        }
+
+        if slug:
+            if video_type == 'events':
+                return self.url_result(
+                    self._search_regex(r'<meta\s+name="externalVideoUrl"\s+content="([^"]+)"', webpage, 'videoUrl'), url_transparent=True, **metainfo, **{
+                        'timestamp': parse_iso8601(self._search_regex(
+                            r'<meta\s+name="startDate"\s+content="([^"]+)"', webpage, 'date', default=None)),
+                    })
+            else:
+                entry_id = self._search_regex(r'<meta name="entryId" content="([^"]+)"', webpage, 'entryId')
+                video_info = self._download_json(
+                    f'https://learn.microsoft.com/api/video/public/v1/entries/{entry_id}', video_id)
+                return {
+                    'id': entry_id,
+                    'formats': self._extract_ism(video_info['publicVideo']['adaptiveVideoUrl'], video_id),
+                    'subtitles': self._sub_to_dict(traverse_obj(video_info, ('publicVideo', 'captions', ..., {
+                        'tag': ('language', {str}),
+                        'url': ('url', {url_or_none}),
+                    }))),
+                    **metainfo,
+                    **traverse_obj(video_info, {
+                        'timestamp': ('createTime', {parse_iso8601}),
+                        'thumbnails': ('publicVideo', 'thumbnailOtherSizes', ..., {lambda x: {'url': x}}),
+                    }),
+                }
+        else:
+            url_base = f'https://learn.microsoft.com/api/contentbrowser/search/{video_type}/{series}/{"sessions" if video_type == "events" else "episodes"}?locale=en-us'
```

Comment:
```suggestion
            url_base = f'https://learn.microsoft.com/api/contentbrowser/search/{video_type}/{series}/{"sessions" if video_type == "events" else "episodes"}'
```
refer to the other comment